Origin: EmberBSD, AI-assisted.
pkgsrc/versions.mk keeps Rails 8.1 only; the EmberBSD recipes add the
Rails 8.1.4 packages and www/ruby-action_text-trix. rails.mk offers,
accepts and defaults to 81, and a package that accepts only an older
Rails gets a fail reason instead of a parse error. Move the packages that
accepted 7.2 or 8.0 onto 8.1: metasploit-model 5.0.6 and
metasploit-concern 5.0.7 allow Rails < 9; repofetch 0.5.2 and xdr 3.0.3,
the last releases, have their Rails upper bounds relaxed with
OVERRIDE_GEMSPEC. Bump PKGREVISION where the Rails dependency changes and
the package name does not.
Not submitted upstream.

--- a/databases/ruby-activeldap/Makefile
+++ b/databases/ruby-activeldap/Makefile
@@ -17,7 +17,7 @@
 
 USE_LANGUAGES=	# none
 
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/databases/ruby-activerecord-odbc/Makefile
+++ b/databases/ruby-activerecord-odbc/Makefile
@@ -2,7 +2,7 @@
 
 DISTNAME=	activerecord-odbc-adapter-2.0
 PKGNAME=	${RUBY_PKGPREFIX}-${DISTNAME:S/-adapter//}
-PKGREVISION=	4
+PKGREVISION=	5
 CATEGORIES=	databases
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/databases/ruby-arel-helpers/Makefile
+++ b/databases/ruby-arel-helpers/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.14 2026/01/17 15:39:33 taca Exp $
 
 DISTNAME=	arel-helpers-2.17.0
+PKGREVISION=	1
 CATEGORIES=	databases
 
 MAINTAINER=	minskim@NetBSD.org
--- a/databases/ruby-awesome_nested_set/Makefile
+++ b/databases/ruby-awesome_nested_set/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.23 2026/01/13 14:50:59 taca Exp $
 
 DISTNAME=	awesome_nested_set-3.9.0
+PKGREVISION=	1
 CATEGORIES=	databases
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/databases/ruby-with_advisory_lock/Makefile
+++ b/databases/ruby-with_advisory_lock/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.2 2026/08/11 15:42:54 taca Exp $
 
 DISTNAME=	with_advisory_lock-7.6.0
+PKGREVISION=	1
 CATEGORIES=	databases
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -15,7 +16,7 @@
 #
 # This pacakge is for redmine60 and redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
--- a/devel/ruby-gettext-activerecord/Makefile
+++ b/devel/ruby-gettext-activerecord/Makefile
@@ -2,7 +2,7 @@
 
 DISTNAME=	gettext_activerecord-2.1.0
 PKGNAME=	${RUBY_PKGPREFIX}-${DISTNAME:S/_/-/}
-PKGREVISION=	4
+PKGREVISION=	5
 CATEGORIES=	devel
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/devel/ruby-gettext-rails/Makefile
+++ b/devel/ruby-gettext-rails/Makefile
@@ -2,7 +2,7 @@
 
 DISTNAME=	gettext_rails-2.1.0
 PKGNAME=	${RUBY_PKGPREFIX}-${DISTNAME:S/_/-/}
-PKGREVISION=	3
+PKGREVISION=	4
 CATEGORIES=	devel
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/devel/ruby-globalid/Makefile
+++ b/devel/ruby-globalid/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.19 2026/06/28 14:36:34 taca Exp $
 
 DISTNAME=	globalid-1.4.0
+PKGREVISION=	1
 CATEGORIES=	devel
 
 MAINTAINER=	minskim@NetBSD.org
--- a/devel/ruby-protobuf-cucumber/Makefile
+++ b/devel/ruby-protobuf-cucumber/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.4 2023/09/03 15:42:31 taca Exp $
 
 DISTNAME=	protobuf-cucumber-3.10.8
-PKGREVISION=	2
+PKGREVISION=	3
 CATEGORIES=	devel
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/devel/ruby-rspec-rails/Makefile
+++ b/devel/ruby-rspec-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.59 2026/03/12 15:48:07 taca Exp $
 
 DISTNAME=	rspec-rails-8.0.4
+PKGREVISION=	1
 CATEGORIES=	devel
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -18,7 +19,7 @@
 
 USE_LANGUAGES=	# empty
 
-RUBY_RAILS_ACCEPTED=	72 80
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/devel/ruby-shoulda-matchers/Makefile
+++ b/devel/ruby-shoulda-matchers/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.29 2026/06/28 14:50:40 taca Exp $
 
 DISTNAME=	shoulda-matchers-8.0.1
+PKGREVISION=	1
 CATEGORIES=	devel
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -11,7 +12,7 @@
 DEPENDS+=	${RUBY_ACTIVESUPPORT_DEPENDS}
 
 USE_LANGUAGES=		# none
-RUBY_RAILS_ACCEPTED=	72 80
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/finance/ruby-activemerchant/Makefile
+++ b/finance/ruby-activemerchant/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.21 2024/09/18 15:36:56 taca Exp $
 
 DISTNAME=	activemerchant-1.137.0
+PKGREVISION=	1
 CATEGORIES=	finance
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/lang/ruby/rails.mk
+++ b/lang/ruby/rails.mk
@@ -9,8 +9,8 @@
 # RUBY_RAILS_DEFAULT
 #	Select default Ruby on Rails version.
 #
-#	Possible values: 72 80
-#	Default: 72
+#	Possible values: 81
+#	Default: 81
 #
 #
 # === Infrastructure variables ===
@@ -27,7 +27,7 @@
 # RUBY_RAILS_ACCEPTED
 #	The Ruby on Rails versions that are acceptable for the package.
 #
-#	Possible values: 72 80
+#	Possible values: 81
 #	Default: (empty)
 #
 # RUBY_RAILS_STRICT_DEP
@@ -41,21 +41,20 @@
 # RUBY_RAILS
 #	Selected Ruby on Rails version.
 #
-#	Possible values: 72 80
+#	Possible values: 81
 #
 
 #
 # current Ruby on Rails versions.
 #
-RUBY_RAILS72_VERSION?=	7.2.3.2
-RUBY_RAILS80_VERSION?=	8.0.5.1
+RUBY_RAILS81_VERSION?=	8.1.4
 
 RUBY_RAILS_ACCEPTED?=	# empty
-RUBY_RAILS_DEFAULT?=	72
+RUBY_RAILS_DEFAULT?=	81
 
 RUBY_RAILS_STRICT_DEP?=	no
 
-RUBY_RAILS_SUPPORTED=	72 80
+RUBY_RAILS_SUPPORTED=	81
 
 .if empty(RUBY_RAILS_SUPPORTED:M${RUBY_RAILS_DEFAULT})
 .  error Unsupported RUBY_RAILS_DEFAULT: ${RUBY_RAILS_DEFAULT}
@@ -65,6 +64,15 @@
 RUBY_RAILS_ACCEPTED=	${RUBY_RAILS_SUPPORTED}
 .endif
 
+# EmberBSD keeps only Rails 8.1 (pkgsrc/versions.mk). A package that accepts
+# only an older Rails fails instead of selecting it.
+_RUBY_RAILS_LISTED:=	${RUBY_RAILS_ACCEPTED}
+RUBY_RAILS_ACCEPTED:=	${RUBY_RAILS_SUPPORTED:@rr@${_RUBY_RAILS_LISTED:M${rr}}@}
+.if empty(RUBY_RAILS_ACCEPTED)
+PKG_FAIL_REASON+=	"EmberBSD keeps only Rails ${RUBY_RAILS_SUPPORTED}; ${PKGPATH} accepts ${_RUBY_RAILS_LISTED}."
+RUBY_RAILS_ACCEPTED=	${RUBY_RAILS_DEFAULT}
+.endif
+
 .if defined(RUBY_RAILS_REQD)
 .  if empty(RUBY_RAILS_ACCEPTED:M${RUBY_RAILS_REQD})
 .    error Unsupported RUBY_RAILS_REQD: ${RUBY_RAILS_REQD}
@@ -88,10 +96,8 @@
 
 RUBY_RAILS_REQD?=	${RUBY_RAILS}
 
-.if ${RUBY_RAILS} == "80"
-RAILS_VERSION:=	${RUBY_RAILS80_VERSION}
-.elif ${RUBY_RAILS} == "72"
-RAILS_VERSION:=	${RUBY_RAILS72_VERSION}
+.if ${RUBY_RAILS} == "81"
+RAILS_VERSION:=	${RUBY_RAILS81_VERSION}
 .endif
 
 #
--- a/mail/ruby-roadie-rails/Makefile
+++ b/mail/ruby-roadie-rails/Makefile
@@ -14,7 +14,7 @@
 
 USE_LANGUAGES=	# none
 
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/misc/ruby-sprockets-rails/Makefile
+++ b/misc/ruby-sprockets-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.12 2024/09/18 15:40:07 taca Exp $
 
 DISTNAME=	sprockets-rails-3.5.2
+PKGREVISION=	1
 CATEGORIES=	misc
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/net/ruby-xdr/Makefile
+++ b/net/ruby-xdr/Makefile
@@ -1,7 +1,7 @@
 # $NetBSD: Makefile,v 1.8 2023/09/03 15:42:32 taca Exp $
 
 DISTNAME=	xdr-3.0.3
-PKGREVISION=	1
+PKGREVISION=	2
 CATEGORIES=	net
 
 MAINTAINER=	minskim@NetBSD.org
@@ -14,6 +14,8 @@
 
 USE_LANGUAGES=	# none
 
+OVERRIDE_GEMSPEC+=	activemodel>=4.2 activesupport>=4.2
+
 RUBY_RAILS_ACCEPTED=	# empty
 
 .include "../../lang/ruby/gem.mk"
--- a/security/ruby-metasploit-concern/Makefile
+++ b/security/ruby-metasploit-concern/Makefile
@@ -1,6 +1,6 @@
 # $NetBSD: Makefile,v 1.17 2026/04/19 14:18:00 taca Exp $
 
-DISTNAME=	metasploit-concern-5.0.6
+DISTNAME=	metasploit-concern-5.0.7
 CATEGORIES=	security
 
 MAINTAINER=	minskim@NetBSD.org
@@ -11,7 +11,7 @@
 DEPENDS+=	${RUBY_ACTIVEMODEL_DEPENDS}
 DEPENDS+=	${RUBY_RAILTIES_DEPENDS}
 
-RUBY_RAILS_ACCEPTED=	72 80
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/security/ruby-metasploit-concern/distinfo
+++ b/security/ruby-metasploit-concern/distinfo
@@ -1,5 +1,5 @@
 $NetBSD: distinfo,v 1.16 2026/04/19 14:18:00 taca Exp $
 
-BLAKE2s (metasploit-concern-5.0.6.gem) = 7a04cdc6907d25c6285f503640accf58d08498699d6385082bd990790b433c61
-SHA512 (metasploit-concern-5.0.6.gem) = 7720e11222bcffad50193e476aff0c225611237eb35a375a5ada245ae5df2819d811f4a5944c574375b72e3699f8843831a319c6546967ffb06dd6a05b4dd17f
-Size (metasploit-concern-5.0.6.gem) = 12800 bytes
+BLAKE2s (metasploit-concern-5.0.7.gem) = 29d970760172f934ae68d22a6aeb19590d82ad6a33a7f773e2bfe2f47d885a9d
+SHA512 (metasploit-concern-5.0.7.gem) = cf24a1f7e1c65fd04bbe69c37adc003bdc08bfa7d9244d11cbcbd43067135bdb05a8a1badcd9ea00dcc3234749a69ac3473b4a740f07d48a6f693dcca8e26571
+Size (metasploit-concern-5.0.7.gem) = 12800 bytes
--- a/security/ruby-metasploit-model/Makefile
+++ b/security/ruby-metasploit-model/Makefile
@@ -1,6 +1,6 @@
 # $NetBSD: Makefile,v 1.14 2026/04/19 14:20:57 taca Exp $
 
-DISTNAME=	metasploit-model-5.0.5
+DISTNAME=	metasploit-model-5.0.6
 CATEGORIES=	security
 
 MAINTAINER=	minskim@NetBSD.org
@@ -11,7 +11,7 @@
 DEPENDS+=	${RUBY_ACTIVEMODEL_DEPENDS}
 DEPENDS+=	${RUBY_RAILTIES_DEPENDS}
 
-RUBY_RAILS_ACCEPTED=	72 80
+RUBY_RAILS_ACCEPTED=	81
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/security/ruby-metasploit-model/distinfo
+++ b/security/ruby-metasploit-model/distinfo
@@ -1,5 +1,5 @@
 $NetBSD: distinfo,v 1.13 2026/04/19 14:20:57 taca Exp $
 
-BLAKE2s (metasploit-model-5.0.5.gem) = ee058d65c4f3b8ecc9c0c5e9c5211d4dd39b0a2d168b4acd5e706d00f541ee3e
-SHA512 (metasploit-model-5.0.5.gem) = 914e5e0ee5333ab25aeb624dd33f9a625a03f268ad9aa375378f7463f9fff9bfa1c8046f9c32b49e2c56ad00b758259c25c82ecb511bf0b04c6e9a9baf81bb78
-Size (metasploit-model-5.0.5.gem) = 76800 bytes
+BLAKE2s (metasploit-model-5.0.6.gem) = c7d8bc3eee28410f5d5c808125dc246513965abe8346db75df7e4a7dea2456a7
+SHA512 (metasploit-model-5.0.6.gem) = 012ea7bc155208642992ffe421fe1049b8cfded9becc10ef7a3c8ca3cf4858f905efa3259cc0df1577d4f2c5abc2a870d1e3ab1f7d4d7675583c90ef0865ce77
+Size (metasploit-model-5.0.6.gem) = 76800 bytes
--- a/sysutils/ruby-repofetch/Makefile
+++ b/sysutils/ruby-repofetch/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.6 2026/02/23 14:27:24 taca Exp $
 
 DISTNAME=	repofetch-0.5.2
+PKGREVISION=	1
 CATEGORIES=	sysutils
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -17,8 +18,9 @@
 
 USE_LANGUAGES=	# none
 
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 RUBYGEM_OPTIONS+=	--format-executable
+OVERRIDE_GEMSPEC+=	actionview>=7.0.4
 
 .include "../../lang/ruby/gem.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/textproc/ruby-rails-dom-testing/Makefile
+++ b/textproc/ruby-rails-dom-testing/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.14 2025/05/21 14:38:28 taca Exp $
 
 DISTNAME=	rails-dom-testing-2.3.0
+PKGREVISION=	1
 CATEGORIES=	textproc
 
 MAINTAINER=	minskim@NetBSD.org
--- a/www/ruby-bullet/Makefile
+++ b/www/ruby-bullet/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.5 2026/09/06 16:13:17 taca Exp $
 
 DISTNAME=	bullet-8.2.0
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/www/ruby-doorkeeper/Makefile
+++ b/www/ruby-doorkeeper/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.1 2025/11/03 05:49:50 taca Exp $
 
 DISTNAME=	doorkeeper-5.8.2
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -13,7 +14,7 @@
 #
 # This pacakge is for redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
--- a/www/ruby-importmap-rails/Makefile
+++ b/www/ruby-importmap-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.2 2026/01/18 15:12:12 taca Exp $
 
 DISTNAME=	importmap-rails-2.2.3
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -15,7 +16,7 @@
 #
 # This pacakge is for redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
--- a/www/ruby-jquery-rails/Makefile
+++ b/www/ruby-jquery-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.21 2025/11/03 06:07:23 taca Exp $
 
 DISTNAME=	jquery-rails-4.6.1
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/www/ruby-propshaft/Makefile
+++ b/www/ruby-propshaft/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.2 2026/04/19 14:41:11 taca Exp $
 
 DISTNAME=	propshaft-1.3.2
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -15,7 +16,7 @@
 #
 # This pacakge is for redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
--- a/www/ruby-requestjs-rails/Makefile
+++ b/www/ruby-requestjs-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.2 2026/01/18 15:36:28 taca Exp $
 
 DISTNAME=	requestjs-rails-0.0.14
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -13,7 +14,7 @@
 #
 # This pacakge is for redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
--- a/www/ruby-sassc-rails/Makefile
+++ b/www/ruby-sassc-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.3 2020/05/21 16:04:24 taca Exp $
 
 DISTNAME=	sassc-rails-2.1.2
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
--- a/www/ruby-stimulus-rails/Makefile
+++ b/www/ruby-stimulus-rails/Makefile
@@ -1,6 +1,7 @@
 # $NetBSD: Makefile,v 1.1 2025/11/03 07:09:02 taca Exp $
 
 DISTNAME=	stimulus-rails-1.3.4
+PKGREVISION=	1
 CATEGORIES=	www
 
 MAINTAINER=	pkgsrc-users@NetBSD.org
@@ -13,7 +14,7 @@
 #
 # This pacakge is for redmine61
 #
-RUBY_RAILS_ACCEPTED=	72
+RUBY_RAILS_ACCEPTED=	81
 
 USE_LANGUAGES=	# none
 
